pkg/analyze/storage_class.go

```diff
@@ -31,7 +31,8 @@ func analyzeStorageClass(analyzer *troubleshootv1beta1.StorageClass, getCollecte
 	}
 
 	for _, storageClass := range storageClasses {
-		if storageClass.Name == analyzer.StorageClassName {
+		val, _ := storageClass.Annotations["storageclass.kubernetes.io/is-default-class"]
+		if (storageClass.Name == analyzer.StorageClassName) || (analyzer.StorageClassName == "" && val == "true") {
```

Comment:
so the way I'm reading this, you could do

```
storageClass:
  name: "gold"
```

to check that a storage class exists with the name `gold`, or you could do 

```
storageClass: {}
```

to check that any default storage class exists. @markpundsack I'm okay with this overloading, but does that work for you, or do you want to do a separate analyzer `hasDefaultStorageClass` or something?

Reply:
Yes, exactly! When a name is specified, it doesn't check the is-default flag, and when no name is provided, it checks if there is at least one default sc. 
I could add a new optional field in the analyzer storageClass specs to check if a storage class with a given name is also marked as default. Something like:
```YAML
- storageClass:
        storageClassName: "my_sc"
        isDefault: true/false 
```

Reply:
I'm OK with overloading since specifying `storageClass: {}` doesn't really have any other reasonable interpretation.

I don't it makes sense to check if a specific storage class is default or not.

Reply:
@markpundsack @dexhorthy I could add a default fail/pass message in absence of an user-defined one. If storageClass were used as ```storageClass: {}``` add a message "There is/is not a default storage class". In addition a more explicative checkName could also be add, something like "Storage Class Marked as Default". What do you think?

Reply:
Adding default message and name makes sense. Not sure about the wording.

Reply:
Messages look good.